src/algokit/core/deploy.py

```diff
@@ -0,0 +1,157 @@
+# 1. User can call algokit deploy to different networks
+# 2. By default algokit cli contains configs for testnet and mainnet
+# 3. User can overwrite them by creating a config file in the project root
+
+
+import configparser
+import logging
+import os
+import sys
+from pathlib import Path
+
+import click
+
+from algokit.core import proc
+from algokit.core.bootstrap import ALGOKIT_CONFIG
+
+if sys.version_info >= (3, 11):
+    import tomllib
+else:
+    import tomli as tomllib
+
+logger = logging.getLogger(__name__)
+
+
+def _generate_config(  # noqa: PLR0913
+    algod_server: str,
+    indexer_server: str,
+    algod_token: str = "",
+    algod_port: str = "",
+    indexer_token: str = "",
+    indexer_port: str = "",
+) -> dict[str, str]:
+    return {
+        "ALGOD_TOKEN": algod_token,
+        "ALGOD_SERVER": algod_server,
+        "ALGOD_PORT": algod_port,
+        "INDEXER_TOKEN": indexer_token,
+        "INDEXER_SERVER": indexer_server,
+        "INDEXER_PORT": indexer_port,
+    }
+
+
+REQUIRED_KEYS = [
+    "ALGOD_TOKEN",
+    "ALGOD_SERVER",
+    "ALGOD_PORT",
+    "INDEXER_TOKEN",
+    "INDEXER_SERVER",
+    "INDEXER_PORT",
+]
+
+DEPLOY_CONFIGS = {
+    "localnet": _generate_config(
+        algod_token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        algod_server="http://localhost",
+        algod_port="4001",
+        indexer_token="aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa",
+        indexer_server="http://localhost",
+        indexer_port="8980",
+    ),
+    "testnet": _generate_config(
```

Comment:
this is duplicating some stuff from `cli/explore.py`, maybe we should refactor here?

Reply:
moved them out to new constants.py that also aggregates a bunch of other env vars reused across core